components/canvas/raqote_backend.rs

```diff
@@ -730,6 +725,28 @@ impl GenericPath<RaqoteBackend> for Path {
         let path: raqote::Path = (&*self).into();
         self.0.set(path.transform(transform).into());
     }
+
+    fn bound_box(&self) -> Rect<f64> {
+        let path: raqote::Path = self.into();
+        let mut points = vec![];
+        for op in path.ops {
+            match op {
+                PathOp::MoveTo(p) => points.push(p),
+                PathOp::LineTo(p) => points.push(p),
+                PathOp::QuadTo(p1, p2) => {
+                    points.push(p1);
+                    points.push(p2);
+                },
+                PathOp::CubicTo(p1, p2, p3) => {
+                    points.push(p1);
+                    points.push(p2);
+                    points.push(p3);
+                },
+                PathOp::Close => {},
+            }
+        }
+        Box2D::from_points(points).to_rect().cast()
+    }
```

Comment:
Can you do this without allocating by simply keeping a minimum point and maximum point and then making a rectangle that includes both? Maybe give that a shot before landing. I think avoiding allocation here makes sense as canvas is often on the hot path.

Reply:
That's essentially what Box2D::from_points is doing, so if I find a way to do this with iterator ... with flattening because each op can make multiple points.